src/3d/qgsskyboxsettings.h

```diff
@@ -0,0 +1,127 @@
+      mIsSkyboxEnabled = element.attribute( QStringLiteral( "skybox-enabled" ) ).toInt();
+      mSkyboxType = element.attribute( QStringLiteral( "skybox-type" ) );
+      mSkyboxBaseName = element.attribute( QStringLiteral( "base-name" ) );
+      mSkyboxExt = element.attribute( QStringLiteral( "extension" ) );
+      mHDRTexturePath = element.attribute( QStringLiteral( "HDR-texture-path" ) );
+      mCubeMapFacesPaths.clear();
+      mCubeMapFacesPaths[ QStringLiteral( "posX" ) ] = element.attribute( QStringLiteral( "posX-texture-path" ) );
+      mCubeMapFacesPaths[ QStringLiteral( "posY" ) ] = element.attribute( QStringLiteral( "posY-texture-path" ) );
+      mCubeMapFacesPaths[ QStringLiteral( "posZ" ) ] = element.attribute( QStringLiteral( "posZ-texture-path" ) );
+      mCubeMapFacesPaths[ QStringLiteral( "negX" ) ] = element.attribute( QStringLiteral( "negX-texture-path" ) );
+      mCubeMapFacesPaths[ QStringLiteral( "negY" ) ] = element.attribute( QStringLiteral( "negY-texture-path" ) );
+      mCubeMapFacesPaths[ QStringLiteral( "negZ" ) ] = element.attribute( QStringLiteral( "negZ-texture-path" ) );
+    }
+
+    //! Writes settings to a DOM \a element
+    void writeXml( QDomElement &element, const QgsReadWriteContext &context ) const
+    {
+      Q_UNUSED( context );
+      element.setAttribute( QStringLiteral( "skybox-enabled" ), mIsSkyboxEnabled );
+      element.setAttribute( QStringLiteral( "skybox-type" ), mSkyboxType );
+      element.setAttribute( QStringLiteral( "base-name" ), mSkyboxBaseName );
+      element.setAttribute( QStringLiteral( "extension" ), mSkyboxExt );
+      element.setAttribute( QStringLiteral( "HDR-texture-path" ), mHDRTexturePath );
+      element.setAttribute( QStringLiteral( "posX-texture-path" ), mCubeMapFacesPaths[ QStringLiteral( "posX" ) ] );
+      element.setAttribute( QStringLiteral( "posY-texture-path" ), mCubeMapFacesPaths[ QStringLiteral( "posY" ) ] );
+      element.setAttribute( QStringLiteral( "posZ-texture-path" ), mCubeMapFacesPaths[ QStringLiteral( "posZ" ) ] );
+      element.setAttribute( QStringLiteral( "negX-texture-path" ), mCubeMapFacesPaths[ QStringLiteral( "negX" ) ] );
+      element.setAttribute( QStringLiteral( "negY-texture-path" ), mCubeMapFacesPaths[ QStringLiteral( "negY" ) ] );
+      element.setAttribute( QStringLiteral( "negZ-texture-path" ), mCubeMapFacesPaths[ QStringLiteral( "negZ" ) ] );
+    }
+
+    //! Returns whether the skybox is enabled
+    bool isSkyboxEnabled() const { return mIsSkyboxEnabled; }
+    //! Sets whether the skybox is enabled
+    void setIsSkyboxEnabled( bool enabled ) { mIsSkyboxEnabled = enabled; }
+
+    /**
+     * Returns the type of the skybox
+     * This can be one of 3 types: "Textures collection", "HDR texture", "Distinct Faces"
+     */
+    QString skyboxType() const { return mSkyboxType; }
+
+    /**
+     * Sets the type of the skybox
+     * the type can be one of 3 types: "Textures collection", "HDR texture", "Distinct Faces"
```

Comment:
It would be cleaner to use an enum for these types, instead of the string values